src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/streaming_endpoint.py

```diff
@@ -64,6 +66,66 @@ def remove_akamai_access_control(client, account_name, resource_group_name, stre
     return client.update(resource_group_name, account_name, streaming_endpoint_name, streaming_endpoint)
 
 
+def update_streaming_endpoint_setter(client, resource_group_name, account_name, streaming_endpoint_name,
+                             parameters):
+    if parameters.access_control is not None:
+        if parameters.access_control.ip is not None:
+            ips = list(map(lambda x: create_ip_range(streaming_endpoint_name, x) if isinstance(x, str) else x,
+                           parameters.access_control.ip.allow))
+        else:
+            raise CLIError('The streaming endpoint resource does not have an IP allow list, you must initialize it.')
+    else:
+            raise CLIError('The streaming endpoint resource does not have an Akamai access control, set it before using -add parameter.')
+
+    parameters.access_control.ip.allow = ips
+    return client.update(resource_group_name, account_name, streaming_endpoint_name, parameters)
+
+
+def update_streaming_endpoint(instance, tags=None, cross_domain_policy=None, client_access_policy=None,
+                              description=None, max_cache_age=None, ips=None,
+                              cdn_provider=None, cdn_profile=None, custom_host_names=None):
+    from azure.mgmt.media.models import (IPAccessControl, StreamingEndpointAccessControl)
+
+    if not instance:
+        raise CLIError('The streaming endpoint resource was not found.')
+
+    if ips is not None and (cdn_provider is not None or cdn_profile is not None):
```

Comment:
This will be validated by the API .. is it necessary?

Reply:
No, it isn't, you're right